label_studio/core/middleware.py

```diff
@@ -248,3 +249,42 @@ def process_response(self, request, response):
                 del response['Content-Security-Policy-Report-Only']
             delattr(response, '_override_report_only_csp')
         return response
+
+
+def get_user_jwt(request):
+    from django.contrib.auth.middleware import get_user
+    from rest_framework_simplejwt.authentication import JWTAuthentication
+
+    try:
+        user = get_user(request)
+        if user.is_authenticated:
+            return user
+
+        jwt_authentication = JWTAuthentication()
+        auth_header = jwt_authentication.get_header(request)
+        if not auth_header:
+            return None
+        if isinstance(auth_header, str):
+            auth_header = auth_header.encode()
+
+        raw_token = jwt_authentication.get_raw_token(auth_header)
+        validated_token = jwt_authentication.get_validated_token(
+            raw_token
+        )
+        user = jwt_authentication.get_user(validated_token)
+    except:
```

Comment:
seems like a lot of different cases to cover under one try-except - what exceptions would we be catching here?

Reply:
So basically my thinking here is that, if we fail to authenticate, just return no user (which means we don't set the "is_jwt" attribute on the request). Granted, it at least makes sense to log what the actual exception is here, which I'll go ahead and add.

What I've seen trigger this for the most part, though, is requests that are using something other than JWT for their auth token (e.g. the old api tokens and SCIM)

Reply:
make sure to make it clear in the logs that this is normal